lib/puppet_x/puppetlabs/powershell/powershell_manager.rb

```diff
@@ -75,8 +75,10 @@ def initialize(cmd, debug, pipe_timeout)
           # clutter behind, so explicitly close those too.
           @stdout.close if !@stdout.closed?
           @stderr.close if !@stderr.closed?
-          Process.kill("KILL", @ps_process[:pid]) if @ps_process.alive?
-          raise "Failure waiting for PowerShell process #{@ps_process[:pid]} to start pipe server"
+          kill_called = Process.kill("KILL", @ps_process[:pid]) if @ps_process.alive?
```

Comment:
So I don't think the return value here means kill called... see https://github.com/ruby/ruby/blob/trunk/signal.c#L413-L532. In particular, the return is `INT2FIX(i-1);`, which Ruby docs don't even document. http://www.rubydoc.info/stdlib/core/Process.kill